### ISSUE TITLE
Use GPU in calculating thickness 

### ISSUE THREAD

**Issue by imadharilla** [mikedh/trimesh#1141]
Hello, 
first of all, thank you so much for such an amazing library! 

I'm calculating the thickness of every face in the mesh and trying to color the mesh and display it. 
The problem is such operation : `thick = trimesh.proximity.thickness(mesh=m, points=points)` is compute-intensive, and I believe that using GPU for such type of computation will be a huge jump in term of performance. 

Wondering if there is a way to use cuda or with trimesh functions to benefit from the GPU ?

**Comment by mikedh** [comment]
Yeah that could be cool! It's also possible a better C implementation, maybe using [igl](https://libigl.github.io/libigl-python-bindings/igl_docs/#signed_distance) could be a LOT faster.
